samples/FunctionApp/Program.cs

```diff
@@ -21,9 +23,17 @@ static async Task Main(string[] args)
                 .ConfigureFunctionsWorkerDefaults(builder =>
                 {
                     builder
-                        .AddApplicationInsights()
+                        .AddApplicationInsights(options =>
+                        {
+                            // Configure the underlying ApplicationInsightsServiceOptions
+                            options.EnableAdaptiveSampling = false;
+                        })
                         .AddApplicationInsightsLogger();
                 })
+                // The Application Insights SDK adds a default logging filter that instructs ILogger to capture only Warning and more severe logs. Application Insights requires an explicit override.
+                // Log levels can also be configured using appsettings.json. For more information, see https://learn.microsoft.com/en-us/azure/azure-monitor/app/worker-service#ilogger-logs
+                .ConfigureLogging(logging => logging
```

Comment:
I prefer removing the app insights filter as opposed to updating it. The difference is this now lets app insights take on default log filters:

``` CSharp
services.Configure<LoggerFilterOptions>(options =>
            {
                LoggerFilterRule toRemove = options.Rules.FirstOrDefault(rule => rule.ProviderName
                    == "Microsoft.Extensions.Logging.ApplicationInsights.ApplicationInsightsLoggerProvider");

                if (toRemove is not null)
                {
                    options.Rules.Remove(toRemove);
                }
            });
```

Removing the filter will allows the general `Logging:LogLevel` filter values to apply to app insights. Without removing it you have to explicitly configure `Logging:ApplicationInsights:LogLevel`.